### PR TITLE
Client list_models integration

### PR DESCRIPTION
**Pull request by cliveseldon**

### CODE REVIEW

File: tempo/kfserving/k8s.py [hunk 1]
```diff
@@ -242,3 +258,49 @@ def _get_spec(self, model_spec: ModelSpec) -> dict:
...
+                remote_model = deserialize(json.loads(metadata))
+                models.append(remote_model)
+            return models
+        except ApiException as e:
+            if e.status == 404:
+                return []
+            else:
+                raise e
+
+    # def load_remote(self, model_listing: KubernetesModelListing) -> RemoteModel:
```
Comment (axsaucedo) [inline tempo/kfserving/k8s.py:292]: Remove

File: tempo/kfserving/k8s.py [hunk 2]
```diff
@@ -242,3 +258,49 @@ def _get_spec(self, model_spec: ModelSpec) -> dict:
...
+            namespace = self.runtime_options.k8s_options.namespace
+
+        if namespace is None:
+            return []
+
+        try:
+            models = []
+            response = api_instance.list_namespaced_custom_object(
+                group="serving.kubeflow.org",
+                version="v1beta1",
```
Comment (axsaucedo) [inline tempo/kfserving/k8s.py:276]: Would it be limited to just v1beta1? We could add issue to extend to the earlier crds

File: tempo/serve/runtime.py [hunk 3]
```diff
@@ -15,8 +17,20 @@ class ModelSpec(BaseModel):
...
+    def ensure_type(cls, v):
+        if isinstance(v, str):
+            klass = locate(v)
+            return klass()
+        else:
+            return v
+
     class Config:
         arbitrary_types_allowed = True
+        json_encoders = {
```
Comment (axsaucedo) [inline tempo/serve/runtime.py:30]: For this we could use the pydantic type for classes/modules

File: tempo/serve/stub.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from ..conf import settings
+from ..utils import logger
+
+
+def _load_rclone_cfg() -> str:
+    with open(settings.rclone_cfg, "r") as f:
+        return f.read()
+
+
+def deserialize(d: dict) -> RemoteModel:
```
Comment (axsaucedo) [inline tempo/serve/stub.py:20]: deserialize sounds like it will load from pickle, perhaps we could just call it model_from_spec?

Comment (axsaucedo) [reply tempo/serve/stub.py:20]: Well I guess serialize / deserialize is actually correect, so we can probably stick with this

File: tempo/kfserving/k8s.py (hunk 2 above)
Comment (cliveseldon) [reply tempo/kfserving/k8s.py:276]: yes at present

File: tempo/serve/stub.py (hunk 4 above)
Comment (cliveseldon) [reply tempo/serve/stub.py:20]: ok

**Pull request merged**